doc(stoneintg-756): write how to manually re-trigger integration test (#236)
redhat-appstudio/docs.appstudio.io: docs/modules/ROOT/pages/how-to-guides/testing_applications/proc_retriggering_integration_tests.adoc

= Retriggering Integration Tests

Integration test scenarios for a given snapshot can be re-triggered by adding a label to the snapshot.

== Prerequisites

> - The snapshot for which tests will be retriggered has completed its initial tests

Reviewer: ```suggestion
- The snapshot must complete all its initial tests before you trigger subsequent tests.
```

Author: I tweaked this suggestion a tiny bit, let me know if its fine this way.

> - CLI access to the OpenShift cluster in question.  For information on getting access to the CLI, see xref:../../getting-started/getting_started_in_cli.adoc[Getting started in CLI]

Reviewer: ```suggestion
- Ensure you have CLI access to the specific OpenShift cluster. For information on obtaining CLI access, refer to  xref:../../getting-started/getting_started_in_cli.adoc[Getting started in CLI]
```

- An up-to-date kubectl binary.  The `oc` binary may be used as well.

== Procedure

> . Determine which IntegrationTestScenario should be re-run.  For a given snapshot, only one scenario can

Reviewer: ```suggestion
. Identify the IntegrationTestScenario that needs to be rerun.  For a given snapshot, you can rerun only one IntegrationTestScenario
```

> be re-run at a time.

Reviewer: ```suggestion
be re-run at a time.
```

> . Add the label `test.appstudio.openshift.io/run` to the snapshot. Set the value to the name of the scenario to be re-run

Reviewer: ```suggestion
. Label the snapshot with `test.appstudio.openshift.io/run`, assigning the scenario name you wish to rerun as its value.
```

[source]
$ kubectl label [snapshot name] test.appstudio.openshift.io/run=[scenario name]

> . The tests will be re-triggered automatically.  Once they are re-triggered the label will be removed.  At this point you

Reviewer: ```suggestion
. The tests are re-triggered automatically.  Once they are re-triggered, the system removes the label, allowing you to apply a new label for a different scenario if you wish to test multiple scenarios.
```

> may add the label again with a different scenario if you wish to re-test multiple scenarios.

Reviewer: ```suggestion

```

[source]
$ tkn pipelinerun list                                                                                                                                                          main 

> NAME                            STARTED         DURATION   STATUS

Reviewer: ```suggestion

[Example Output]
NAME                            STARTED         DURATION   STATUS
```

integration-pipelinerun-jfrdb   4 seconds ago   ---        Running

> . If you run `tkn pipelinerun describe [pipelinerun]` you can see that the `appstudio.openshift.io/snapshot` and `test.appstudio.openshift.io/scenario` labels reference the correct snapshot and scenario.

Reviewer: ```suggestion
. To verify the tests, run `tkn pipelinerun describe [pipelinerun]` to ensure that the labels `appstudio.openshift.io/snapshot` and `test.appstudio.openshift.io/scenario` correctly reference the correct snapshot and scenario.
```

Author: I tweaked this suggestion a bit, let me know if its fine this way.

[source]
$ tkn pipelinerun describe integration-pipelinerun-jfrdb                                                                                                                        main 

> Name:              integration-pipelinerun-jfrdb

Reviewer: ```suggestion

[Expected labels in the output]
Name:              integration-pipelinerun-jfrdb
```
